models/base/document.py

```diff
@@ -148,9 +176,33 @@ def __unlink_referee(self, referee):
                     documents.remove(referee)
         self.save()
 
+    def _get_root_document(self):
+        """
+        Get the root-level document.
+        """
+        for referrer in self._referenced_by:
+            root = referrer._get_root_document()
+            if root is not None:
+                return root
+
     def __str__(self):
         return f'{self.__class__.__name__}({self._data})'
 
+    def __getstate__(self):
+        """
+        Customise pickling: referrers are not pickled to avoid circular references.
```

Comment:
Do you mean 'Custom pickling' or 'Customised pickling'? I'm unsure of this current phrasing.

Reply:
The `pickle` will look for a `__getstate__` method to call while pickling an object, before using the default pickling implementation. So the existence of this method is to customise the pickling process.